x-pack/plugins/observability_solution/observability/public/pages/slo_details/components/overview/synthetics_indicator_overview.tsx

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiBadge, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { syntheticsAvailabilityIndicatorSchema, SLOWithSummaryResponse } from '@kbn/slo-schema';
+import React from 'react';
+import { useKibana } from '../../../../utils/kibana_react';
+import { OverviewItem } from './overview_item';
+import { syntheticsMonitorDetailLocatorID } from '../../../../../common';
+
+interface Props {
+  slo: SLOWithSummaryResponse;
+}
+
+export function SyntheticsIndicatorOverview({ slo }: Props) {
+  const {
+    share: {
+      url: { locators },
+    },
+  } = useKibana().services;
+
+  const locator = locators.get(syntheticsMonitorDetailLocatorID);
+
+  const { 'monitor.name': name, 'observer.geo.name': location } = slo.groupings;
```

Comment:
Short answer: No, it is not possible for name and location to be undefined.

Long answer: Anything is possible

Name is required for heartbeat to run, so any monitor without a name would never be executed and never make it to summary documents.

Location name, too, is a required value for the synthetics app to function properly.

So it is possible for these to be undefined, but it would indicate that the entire system is broken in a major way, and it would likely be caught by tests, either Kibana tests, elastic-agent tests, or synthetics tests.

Reply:
Sorry I meant if we don't set groupBy to `["monitor.name", "observer.geo.name"]`, e.g. it is possible that we don't group by these fields if we use the API directly.

Reply:
Ah. Excellent point! Thank you that is a great point. I'll address.

Reply:
Thanks for addressing